Px.Utils/Validation/DatabaseValidation/DatabaseValidator.cs

```diff
@@ -0,0 +1,398 @@
+﻿using Px.Utils.Exceptions;
+using Px.Utils.PxFile;
+using Px.Utils.PxFile.Metadata;
+using Px.Utils.Validation.SyntaxValidation;
+using System.Collections.Concurrent;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Px.Utils.Validation.DatabaseValidation
+{
+    /// <summary>
+    /// Validates a whole px file database including all px files it contains.
+    /// </summary>
+    /// <param name="directoryPath">Path to the database root directory</param>
+    /// <param name="syntaxConf"><see cref="PxFileSyntaxConf"/> object that defines the tokens and symbols for px file syntax.</param>
+    /// <param name="customPxFileValidators">Optional custom <see cref="IDatabaseValidator"/> validator functions ran for each px file within the database</param>
+    /// <param name="customAliasFileValidators">Optional custom <see cref="IDatabaseValidator"/> validator functions that are ran for each alias file within the database</param>
+    /// <param name="customDirectoryValidators">Optional custom <see cref="IDatabaseValidator"/> validator functions for each subdirectory within the database.</param>
+    /// <param name="fileSystem">Optional <see cref="IFileSystem"/> that defines the file system used for the validation process. Default file system is used if none provided.</param>
+    public class DatabaseValidator(
+        string directoryPath, 
+        PxFileSyntaxConf? syntaxConf = null,
+        IDatabaseValidator[]? customPxFileValidators = null,
+        IDatabaseValidator[]? customAliasFileValidators = null,
+        IDatabaseValidator[]? customDirectoryValidators = null,
+        IFileSystem? fileSystem = null
```

Comment:
I would make the filesystem parameter mandatory. No point in setting defaults for something this critical for the functionality.